libqtile/widget/memory.py

```diff
@@ -0,0 +1,32 @@
+import re
+from . import base
+
+
+def get_meminfo():
+    val = {}
+    with open('/proc/meminfo') as file:
+        for line in file:
+            key, tail = line.split(':')
+            uv = tail.split()
+            val[key] = int(int(uv[0]) / 1000)
+    val['MemUsed'] = val['MemTotal'] - val['MemFree']
+    return val
+
+
+class Memory(base.InLoopPollText):
+    """Displays a memory usage graph."""
+    orientations = base.ORIENTATION_HORIZONTAL
+    defaults = [
+        ("format", "%{MemUsed}M/%{MemTotal}M", "see /proc/meminfo for fields")
+    ]
+
+    def __init__(self, **config):
+        super(Memory, self).__init__(**config)
+        self.add_defaults(Memory.defaults)
+
+    def poll(self):
+        mem = get_meminfo()
+
+        def replace(m):
+            return str(mem.get(m.group(1), "(nil)"))
+        return re.sub("%{([^}]+)}", replace, self.format)
```

Comment:
What exactly is this doing? Is this not something that you can do with `.format()` more easily? (cf. https://docs.python.org/3/library/string.html#format-examples)